examples/threshold_enc.rs

```diff
@@ -0,0 +1,168 @@
+    // Creates a new `SecretSociety`.
+    //
+    // # Arguments
+    //
+    // `n_actors` - the number of operatives in the secret society.
+    // `threshold` - the number of operatives that must collaborate in in order to successfully
+    // decrypt a message must exceed this `threshold`.
+    fn new(n_actors: usize, threshold: usize) -> Self {
+        let mut rng = rand::thread_rng();
+        let sk_set = SecretKeySet::random(threshold, &mut rng).unwrap();
+        let pk_set = sk_set.public_keys();
+
+        let actors = (0..n_actors)
+            .map(|id| {
+                let sk_share = sk_set.secret_key_share(id).unwrap();
+                let pk_share = pk_set.public_key_share(id);
+                Actor::new(id, sk_share, pk_share)
+            })
+            .collect();
+
+        SecretSociety { actors, pk_set }
+    }
+
+    // The secret society publishes its public-key to a publicly accessible key server.
+    fn publish_public_key(&self) -> PublicKey {
+        self.pk_set.public_key()
+    }
+
+    fn get_actor(&mut self, id: usize) -> &mut Actor {
+        self.actors
+            .get_mut(id)
+            .expect("No `Actor` exists with that ID")
+    }
+
+    // Starts a new meeting of the secret society. Each time the set of actors receive an encrypted
+    // message, at least 2 of them (i.e. 1 more than the threshold) must work together to decrypt
+    // the ciphertext.
+    fn start_decryption_meeting(&self) -> DecryptionMeeting {
+        DecryptionMeeting {
+            pk_set: self.pk_set.clone(),
+            ciphertext: None,
+            dec_shares: BTreeMap::new(),
+        }
+    }
+}
+
+// A member of the secret society.
+#[derive(Clone, Debug)]
+struct Actor {
+    id: usize,
+    sk_share: SecretKeyShare,
+    pk_share: PublicKeyShare,
+    msg_inbox: Option<Ciphertext>,
+}
+
+impl Actor {
+    fn new(id: usize, sk_share: SecretKeyShare, pk_share: PublicKeyShare) -> Self {
+        Actor {
+            id,
+            sk_share,
+            pk_share,
+            msg_inbox: None,
+        }
+    }
+}
+
+// Sends an encrypted message to an `Actor`.
+fn send_msg(actor: &mut Actor, enc_msg: Ciphertext) {
+    actor.msg_inbox = Some(enc_msg);
+}
+
+// A meeting of the secret society. At this meeting, actors collaborate to decrypt a shared
+// ciphertext.
+struct DecryptionMeeting {
+    pk_set: PublicKeySet,
+    ciphertext: Option<Ciphertext>,
+    dec_shares: BTreeMap<usize, DecryptionShare>,
+}
+
+impl DecryptionMeeting {
+    // An actor contributes their decryption share to the decryption process.
+    fn accept_decryption_share(&mut self, actor: &mut Actor) {
+        let enc_msg = actor.msg_inbox.take().unwrap();
+        if self.ciphertext.is_none() {
+            self.ciphertext = Some(enc_msg.clone());
+        }
+        let dec_share = actor.sk_share.decrypt_share(&enc_msg).unwrap();
+        let dec_share_is_valid = actor.pk_share.verify_decryption_share(&dec_share, &enc_msg);
```

Comment:
Maybe it could be made clearer that _other_ actors can verify that the actor who produced that decryption share didn't cheat (e.g. use a different ciphertext). That could be misinterpreted here, since both public and secret key share are part of the actor.

Or we could verify against `self.ciphertext` here, so it would fail if the actor received a different ciphertext?